pydevices/acq400Devices/acq2106_mgtdram8.py

```diff
@@ -0,0 +1,327 @@
+                self.nchans = mds.nchans
+                self.empty_buffers = mds.empty_buffers
+                self.full_buffers = mds.full_buffers
+                self.running = False
+
+            def stop(self):
+                self.running = False
+
+            def run(self):
+                if self.debug:
+                    print("DeviceWorker running")
+                BLOCK_MULTIPLE = acq400_hapi.Acq2106_Mgtdram8.MGT_BLOCK_MULTIPLE
+                MGT_BLOCK_BYTES = acq400_hapi.Acq2106_Mgtdram8.MGT_BLOCK_BYTES
+
+                self.running = True
+                ds = 2
+                nc = MGT_BLOCK_BYTES/ds
+
+                # trigger time out count initialization:
+                rc = acq400_hapi.MgtDramPullClient(self.node_addr)
+                nchan = int(self.nchans)
+                try:
+                    for buf in rc.get_blocks(nchan, ncols=(2**22) / nchan / 4, data_size=ds):
+                        self.full_buffers.put(buf)
+                except socket.timeout as e:
+                    print("Got a timeout.")
+                    err = e.args[0]
+                    # this next if/else is a bit redundant, but illustrates how the
+                    # timeout exception is setup
+
+                    if err == 'timed out':
+                        time.sleep(1)
+                        print (' recv timed out, retry later')
+                        # continue                  else:
+                        print (e)
+                        # break
+                except socket.error as e:
+                    # Something else happened, handle error, exit, etc.
+                    print("socket error", e)
+                    self.full_buffers.put(buf[:self.segment_bytes-toread])
+
+
+    def setChanScale(self,num):
+        chan=self.__getattr__(INPFMT % num)
+        chan.setSegmentScale(MDSplus.ADD(MDSplus.MULTIPLY(chan.COEFFICIENT,MDSplus.dVALUE()),chan.OFFSET))
+
+    def lazy_init(self):
+        try:
+            return self.uut
+        except AttributeError:
+            self.uut = acq400_hapi.Acq2106_Mgtdram8(self.node.data(), monitor=False)
+            return self.uut        
+            
+    def init(self):
+        print('Running init')
+        self.lazy_init()
+
+        trg = self.trig_mode.data()
+
+        # The default case is to use the trigger set by sync_role.
+        if self.trig_mode.data() == 'role_default':
+            self.uut.s0.sync_role = "{} {}".format(self.role.data(), self.freq.data())
+        else:
+            # If the user has specified a trigger.
+            self.uut.s0.sync_role = '{} {} TRG:DX={}'.format(self.role.data(), self.freq.data(), trig_types[trg])
+
+        # Now we set the trigger to be soft when desired.
+        if trg == 'soft':
+            self.uut.s0.transient = 'SOFT_TRIGGER=0'
+        if trg == 'automatic':
+            self.uut.s0.transient = 'SOFT_TRIGGER=1'
+
+        print('Finished init')
```

Comment:
should be wrapped as debug message, i.e. silent in regular mode. there is a Device.debug filed that can be queried or use the _debug method.
self._debug('Finished init')
it can also do formatted strings
self._debug('message %d %s', 1, 'test')